packages/foam-vscode/src/features/filtered-resources.ts

```diff
@@ -0,0 +1,252 @@
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+import {
+  IDataStore,
+  URI,
+  FoamWorkspace,
+  Resource,
+  isAttachment,
+  isPlaceholder,
+  getTitle,
+} from 'foam-core';
+import micromatch from 'micromatch';
+import {
+  FilteredResourcesConfig,
+  FilteredResourcesConfigGroupBy,
+} from '../settings';
+import { getContainsTooltip, getNoteTooltip } from '../utils';
+import { dirname, join } from 'path';
+import { FoamFeature } from '../types';
+import { commands } from 'vscode';
+
+const feature: FoamFeature = {
+  activate: (context: vscode.ExtensionContext) => {
+    context.subscriptions.push(
+      commands.registerCommand(
+        'foam-vscode.open-placeholder-note',
+        async (uri: vscode.Uri) => {
+          let dir: string;
+
+          if (vscode.workspace.workspaceFolders) {
+            dir = vscode.workspace.workspaceFolders[0].uri.fsPath.toString();
+          }
+
+          if (!dir) {
+            const activeFile = vscode.window.activeTextEditor?.document;
+            dir = activeFile ? dirname(activeFile.uri.fsPath) : null;
+          }
+
+          if (dir) {
+            const path = join(dir, `${uri.path}.md`);
+            await fs.promises.writeFile(path, `# ${uri.path}`);
+            const ur = vscode.Uri.file(path);
+            await vscode.window.showTextDocument(ur, {
+              preserveFocus: false,
+              preview: false,
+            });
+          }
+        }
+      )
+    );
+  },
+};
+
+export default feature;
+
+export class FilteredResourcesProvider
```

Comment:
We should call this `GroupedResourcesTreeDataProvider`

Reply:
I don't think that this class belongs to a feature.
we can put it in a `src/vscode-utils.ts` file or something like that.

Reply:
this class needs to be documented, in how it works and what it needs.
e.g. we need to say that:
- it will create commands in the shape of `foam-vscode.${this.filteredNoteType}-grouped-by-folder` and that they need to be defined in `package.json`
- the IDs of the provider (see comment below) need to be unique
- ...

Reply:
I ended up moving it to `src/utils/grouped-resources-tree-data-provider.ts`.